R/predictLearner.R

```diff
@@ -58,9 +58,41 @@ predictLearner2 = function(.learner, .model, .newdata, ...) {
       .newdata[ns] = mapply(factor, x = .newdata[ns],
          levels = fls, SIMPLIFY = FALSE)
   }
-  p = predictLearner(.learner, .model, .newdata, ...)
+  if ("missings" %nin% getLearnerProperties(.learner))
+    no.na = removeNALines(.newdata)
+  else
+    no.na = list(newdata = .newdata, inserts = FALSE)
+  if (!nrow(no.na$newdata))
+    no.na = list(newdata = .newdata, inserts = FALSE)  # no choice if all lines contain NA
+  p = predictLearner(.learner, .model, no.na$newdata, ...)
   p = checkPredictLearnerOutput(.learner, .model, p)
-  return(p)
+  return(insertLines(p, no.na$inserts))
+}
+
+removeNALines = function(newdata) {
+  namat = is.na(newdata)
+  if (!any(vlapply(namat, any))) {
```

Comment:
Is this check necessary? As far as I can see the code after would do the right thing in this case as well.

Reply:
I was wrong about what format the return of `is.na(data.frame)` has, I'll drop this part.